### PR TITLE
Data annotations

### PR DESCRIPTION
**Pull request by sneha-brainchild**

### CODE REVIEW

File: src/Brainchild.HMS.Core/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace Brainchild.HMS.Core
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Class1.cs:3]: This has to be deleted

File: src/Brainchild.HMS.Core/Models/Billing.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    public int BillingId { get; set; }
+
+    [Required]
+    public DateTime BillingDate { get; set; }
+    
+    public int BookingId { get; set; }
+    public Booking Booking { get; set ;}
+
+    public int PaymentId { get; set; }
+    public Payment Payment { get; set ;}
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Billing.cs:17]: One billing can have multiple payment. Advance or additional charges etc

File: src/Brainchild.HMS.Core/Models/Booking.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public DateTime CheckinDate { get; set; }
+
+    [Required]
+    public DateTime CheckoutDate { get; set; }
+    
+    public int status { get; set; }
+
+    public int IsCancelled { get; set; }
+
+    public DateTime CancelDate { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Booking.cs:29]: Cancelled Date

File: src/Brainchild.HMS.Core/Models/Booking.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    public int NoOfAChildren { get; set; }
+
+    [Required]
+    public DateTime CheckinDate { get; set; }
+
+    [Required]
+    public DateTime CheckoutDate { get; set; }
+    
+    public int status { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Booking.cs:25]: Status s should be capital

File: src/Brainchild.HMS.Core/Models/ChargeType.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+public class ChargeType
+{
+    [Key]
+    public int ChargeTypeId { get; set; }
+
+    [Required]
+    [Column(TypeName = "varchar(1000)")]
+    public string ChargeTypeDescription { get; set; }       
+    
+    public List<Charge> Charges { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/ChargeType.cs:14]: We don't need this here. It's just a type definition

File: src/Brainchild.HMS.Core/Models/Guest.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+using Microsoft.EntityFrameworkCore;
+
+public class Guest
+{
+    [Key]
+    public int GuestId { get; set; }
+
+    [Required]
+    [Column(TypeName = "varchar(1000)")]
+    public string Name { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Guest.cs:11]: Why I am seeing repeated comments. Already asked to change to Guest name.prefix column name with table name where ever it makes sense

File: src/Brainchild.HMS.Core/Models/Payment.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    public int PaymentTypeID { get; set; }
+    public PaymentType PaymentType { get; set; }
+
+    public float PaymentAmount { get; set; }
+
+    public float PaymentAdvance { get; set; }    
+
+
+    public List<Billing> Billings { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Payment.cs:18]: One to one, right? One payment should be associated with one bill

File: src/Brainchild.HMS.Core/Models/PaymentType.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+public class PaymentType
+{
+    [Key]
+    public int PaymentTypeID { get; set; }
+
+    [Required]
+    [Column(TypeName = "varchar(1000)")]
+    public int PaymentTypeDescription { get; set; }
+
+    public List<Payment> Payments { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/PaymentType.cs:14]: No need here

File: src/Brainchild.HMS.Core/Models/RoomBooking.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
+
+public class RoomBooking
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/RoomBooking.cs:4]: This should be an enum.so probably short datatype

File: src/Brainchild.HMS.Data/Class1.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace Brainchild.HMS.Data
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Data/Class1.cs:3]: Remove this

**Review by rohithkrajan — CHANGES_REQUESTED** [review]
Some comments are repeating

File: src/Brainchild.HMS.Core/Models/Guest.cs (hunk 6 above)
Comment (sneha-brainchild) [reply src/Brainchild.HMS.Core/Models/Guest.cs:11]: By mistake, I will correct it

File: src/Brainchild.HMS.Core/Models/ChargeType.cs (hunk 5 above)
Comment (sneha-brainchild) [reply src/Brainchild.HMS.Core/Models/ChargeType.cs:14]: Ok

File: src/Brainchild.HMS.Core/Models/Billing.cs (hunk 2 above)
Comment (sneha-brainchild) [reply src/Brainchild.HMS.Core/Models/Billing.cs:17]: done

File: src/Brainchild.HMS.Core/Models/Billing.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    [Key]
+    public int BillingId { get; set; }
+    
+    [Required]
+    public DateTime BillingDate { get; set; }
+    
+    public int BookingId { get; set; }
+    public Booking Booking { get; set ;}
+
+    public int? HotelID { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Billing.cs:19]: Why is this nullable int? Hotel I'd is always mandatory?

Comment (rohithkrajan) [reply src/Brainchild.HMS.Core/Models/Billing.cs:19]: Why do we need hotelid if booking is linked

File: src/Brainchild.HMS.Core/Models/CurrencyCode.cs [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        [Key]
+        public int CurrencyCodeId { get; set; }
+
+        [Required]
+        [Column(TypeName = "varchar(1000)")]
+        public string CurrencyCountry { get; set; }
+
+        [Required]
+        [Column(TypeName = "varchar(1000)")]
+        public string CurrencyCodes { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/CurrencyCode.cs:20]: No plural required

File: src/Brainchild.HMS.Core/Models/CurrencyCode.cs [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+        public int CurrencyNumber { get; set; } 
+
+        [Column(TypeName = "varchar(1000)")]
+        public string CurrencySymbol { get; set; }           
+
+        public int HotelID { get; set; }
+        public Hotel Hotel { get; set ;}    
+
+        public List<Charge> Charges { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/CurrencyCode.cs:30]: No need of this link

File: src/Brainchild.HMS.Core/Models/Guest.cs [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+    public string GuestEmail { get; set; }
+
+    [Required]
+    [Column(TypeName = "varchar(100)")]
+    public string GuestPhoneNo { get; set; }
+
+    public string GuestCountry { get; set; }
+    
+    public int HotelID { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Guest.cs:28]: Why do we need this? Already linked via booking

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+namespace Brainchild.HMS.Core.Models
+{
+public class Hotel
+{
+    [Key]
+    public int HotelID { get; set; }
+
+    public int TenantID { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:13]: Did you check orchard core db

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    [Column(TypeName = "varchar(50)")]
+    public string HotelPhone { get; set; }
+
+    [Column(TypeName = "varchar(50)")]
+    public string Place { get; set; }
+
+    [Column(TypeName = "varchar(50)")]
+    public string GSTNo { get; set; }
+
+    public List<Billing> Billing { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:35]: Why do we need this

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    public List<Booking> Bookings { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:36]: Why we need this

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+    public List<ChargeType> ChargeTypes { get; set; }
+    public List<Guest> Guests { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:38]: Why we need this

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+    public List<PaymentType> PaymentTypes { get; set; }
+    public List<RoomType> RoomTypes { get; set; }
+    public List<Tax> Taxes { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:41]: Why

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+    public string Place { get; set; }
+
+    [Column(TypeName = "varchar(50)")]
+    public string GSTNo { get; set; }
+
+    public List<Billing> Billing { get; set; }
+    public List<Booking> Bookings { get; set; }
+    public List<ChargeType> ChargeTypes { get; set; }
+    public List<Guest> Guests { get; set; }
+    public List<PaymentType> PaymentTypes { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:39]: Why

File: src/Brainchild.HMS.Core/Models/Hotel.cs [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+    public List<RoomType> RoomTypes { get; set; }
+    public List<Tax> Taxes { get; set; }
+    public List<CurrencyCode> CurrencyCodes { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Hotel.cs:42]: Why

**Review by rohithkrajan — CHANGES_REQUESTED** [review]
- Why so many sub properties added unnecessarily making objects complicated

File: src/Brainchild.HMS.Core/Models/Currency.cs [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Brainchild.HMS.Core.Models
+{
+    public class Currency
+    {
+        [Key]
+        public int CurrencyCodeId { get; set; }
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Currency.cs:12]: This should currencyId

File: src/Brainchild.HMS.Core/Models/Charge.cs [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+{
+public class Charge
+{
+    [Key]
+    public int ChargeId { get; set; }
+
+    [ForeignKey("ChargeTypeId")]   
+    public ChargeType ChargeType { get; set; }    
+
+    [ForeignKey("CurrencyCodeId")]
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Charge.cs:16]: This should be currencyId

File: src/Brainchild.HMS.Core/Models/Room.cs [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+   [ForeignKey("RoomTypeId")]
+    public RoomType RoomType { get; set; }
+
+    [Required]
+    [Column(TypeName = "varchar(50)")]
+    public string RoomNo { get; set; }
+
+    [Column(TypeName = "int")]
+    public RoomStatus RoomStatus { get; set; }
+
```
Comment (rohithkrajan) [inline src/Brainchild.HMS.Core/Models/Room.cs:23]: We need to differentiate with hotelid

**Review by rohithkrajan — CHANGES_REQUESTED** [review]
Some minor comments

**Review by rohithkrajan — CHANGES_REQUESTED** [review]
One last change . sorry

File: src/Brainchild.HMS.Core/Models/Currency.cs [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        [Key]
+        public int CurrencyId { get; set; }
+
+        [Required]
+        [Column(TypeName = "varchar(1000)")]
+        public string CurrencyCountry { get; set; }
+
+        [Required]
+        [Column(TypeName = "varchar(1000)")]
```
[the remark this answers is not in the document]
Comment (sneha-brainchild) [reply src/Brainchild.HMS.Core/Models/Currency.cs:19]: yes it is 3 letter

**Review by rohithkrajan — APPROVED** [review]
Looks good

**Pull request merged**